java-squid/src/main/java/org/sonar/java/ast/parser/ListTreeImpl.java

```diff
@@ -61,10 +72,35 @@ public Kind getKind() {
 
   @Override
   public Iterator<Tree> childrenIterator() {
-    // TODO
-    throw new UnsupportedOperationException();
+    //FIXME SONARJAVA-547 Separators
+    Iterator<Tree> iterator = (Iterator<Tree>) ((Iterable<? extends Tree>) list).iterator();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this unnecessary cast to "Iterator". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1905)